bin/parse-legacy-file

```diff
@@ -5,4 +5,4 @@ set -eu -o pipefail
 # shellcheck source=../lib/utils.sh
 source "$(dirname "$0")/../lib/utils.sh"
 
-resolve_version "$(cat "$1")"
+resolve_version "$(head -n 1 "$1")"
```

Comment:
An `.nvmrc` file can actually include comments anywhere, even inline

```
v22 # an inline comment
```

Therefore I don't think this covers all necessary cases